examples/2d/jeans_instability.jl

```diff
@@ -0,0 +1,100 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the compressible Euler equations
+gamma = 5/3
+equations_euler = CompressibleEulerEquations2D(gamma)
+
+# the initial conditions could also just be defined here
+initial_conditions = Trixi.initial_conditions_jeans_instability
+
+polydeg = 3
+solver_euler = DGSEM(polydeg, flux_hll)
+
+coordinates_min = (0, 0)
+coordinates_max = (1, 1)
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level=4,
+                n_cells_max=10_000)
+
+semi_euler = SemidiscretizationHyperbolic(mesh, equations_euler, initial_conditions, solver_euler)
+
+
+###############################################################################
+# semidiscretization of the hyperbolic diffusion equations
+resid_tol = 1.0e-4
+equations_gravity = HyperbolicDiffusionEquations2D(resid_tol)
+
+solver_gravity = DGSEM(polydeg, flux_lax_friedrichs)
+
+semi_gravity = SemidiscretizationHyperbolic(mesh, equations_gravity, initial_conditions, solver_gravity,
+                                            source_terms=source_terms_harmonic)
+
+
+###############################################################################
+# combining both semidiscretizations for Euler + self-gravity
+parameters = ParametersEulerGravity(background_density=1.5e7, # aka rho0
+                                    gravitational_constant=6.674e-8, # aka G
+                                    cfl=2.4,
+                                    n_iterations_max=1000,
+                                    timestep_gravity=timestep_gravity_erk52_3Sstar!)
+
+semi = SemidiscretizationEulerGravity(semi_euler, semi_gravity, parameters)
+
+
+###############################################################################
+# ODE solvers, callbacks etc.
+tspan = (0.0, 5.0)
+ode = semidiscretize(semi, tspan);
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 100
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+Trixi.pretty_form_repl(::Val{:energy_potential}) = "∑e_potential"
+Trixi.pretty_form_file(::Val{:energy_potential}) = "e_potential"
+
+function Trixi.analyze(::Val{:energy_potential}, du, u_euler, t, semi::SemidiscretizationEulerGravity)
+
+  u_gravity = Trixi.wrap_array(semi.cache.u, semi.semi_gravity)
```

Comment:
What is going on here and why do we need this?

(see also my other comment on `wrap_array`)

Reply:
See above. I'll explain it more when I see whether it is necessary and works after implementing AMR. Please nag me if I don't.

Reply:
OK!

Reply:
I've tried to use consistent names for variables at the interface to ODE solvers and add a motivation and explanation of `wrap_array` in c5b3f4cff9ce45a6e7cd4600db17a5fae2625a45. Does that look okay to you? If not, we should discuss open questions to improve that part.